src/pages/Products/Products.js

```diff
@@ -3,17 +3,39 @@ import { useParams, useNavigate } from 'react-router-dom';
 import { useSelector } from 'react-redux';
 import GridProducts from 'components/Products/GridProducts';
 import { getProducts } from 'store/products/selectors';
-import SectionHeader from 'components/SectionHeader';
 import ButtonSlim from 'components/UI/ButtonSlim';
-import Wrapper from 'components/UI/Wrapper/Wrapper';
-import ButtonSlimContainer from 'components/UI/ButtonSlimContainer';
+import { getCategoriesWithProducts } from 'store/reselectors';
+import PageTitle from '../../components/PageTitle';
 
 const Products = () => {
   const { productCategory } = useParams();
-  const [isActive, setIsActive] = useState(productCategory);
   const navigate = useNavigate();
+  const [isActive, setIsActive] = useState(productCategory);
+  let selectedProducts = [];
 
-  const products = useSelector(getProducts);
+  const catWithProducts = useSelector(getCategoriesWithProducts);
+  const allProducts = useSelector(getProducts);
+
+  switch (productCategory) {
+    case 'seats':
+      selectedProducts = catWithProducts[0].seats;
+      break;
+    case 'beds':
+      selectedProducts = catWithProducts[1].beds;
+      break;
+    case 'office':
+      selectedProducts = catWithProducts[2].office;
+      break;
+    case 'kitchen':
+      selectedProducts = catWithProducts[3].kitchen;
+      break;
+    case 'all':
+      selectedProducts = allProducts;
+      break;
+    default:
+      [];
+      break;
+  }
```

Comment:
When we have complex logic we usually move it to separate function in separate file, for example `utils.js`